### PR TITLE
OME Data model: development process

### PR DESCRIPTION
**Pull request by sbesson**
See https://trello.com/c/0zNp8HvU/167-enumerations-addition

In preparation of the amendment of these instructions for minor schema release, this PR migrates the existing internal documentation to the public contributing documentation. A few edits have been added on top of the initial commit to better fit the documentation set and match the existing release process.

/cc @mtbc @hflynn

### CODE REVIEW

**Comment by hflynn** [comment]
Staging building now

**Comment by hflynn** [comment]
Staged at http://www.openmicroscopy.org/site/support/contributing-staging/data-model-schema.html
Probably worth adding a link to the Model docs - http://www.openmicroscopy.org/site/support/ome-model/developers/index.html#ome-model-development-process seems an obvious place (there's already a general link there but we could be more specific now)

**Comment by hflynn** [comment]
Looks fine to me, I leave @mtbc to comment on the technicalities.

File: contributing/data-model-schema.txt [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+
+The OME-TIFF files require special handling, as they do not have an
+automatic update tool. First, identify them and copy them to the new
+directory::
+
+  find 2015-01 -name '*.ome.tiff'
+  cp 2015-01/set-1-meta-companion/*.ome.tiff 2016-06/set-1-meta-companion/
+
+Next, each OME-TIFF file must be edited to have the schema version
+changed to that of the new release. They are binary files so choice of
```
Comment (mtbc) [inline contributing/data-model-schema.txt:194]: We have a card for this, right? It's horrifying.

File: contributing/data-model-schema.txt [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+previous release's directory, and the multi-file samples in its
+:file:`tubhiswt-*` directories, are upgraded to the new schema using
+:command:`bfconvert` from the updated Bio-Formats repository: in that
+repository use :command:`ant tools` to generate the necessary
+:file:`bioformats_package.jar` Java archive file. The sample files from
+the subdirectories are provided also as compressed "zip" archive files.
+The files in the :file:`bioformats-artificial` subdirectory are
+generated by other Bio-Formats classes. Putting these facts together,
+setting up the new "2016-06” samples folder on squig is easily
+achieved::
```
Comment (mtbc) [inline contributing/data-model-schema.txt]: Some of these rather internal details may need adjusting on the way? I'm not sure how much our public documentation should explain within-VPN details.

**Comment by mtbc** [comment]
Looks like the content is preserved excellently, thank you for this work. I don't mind if updating happens in later PRs.

File: contributing/data-model-schema.txt (hunk 2 above)
Comment (sbesson) [reply contributing/data-model-schema.txt]: Fixed in    bb8b9a3

File: contributing/data-model-schema.txt (hunk 1 above)
Comment (sbesson) [reply contributing/data-model-schema.txt:194]: Agreed we need a remedy although I do not think we have discussed/agreed on a solution. I could either see some API in ImageConverter to preserve the original file layout or a consumer project of OME Files scripting the OME-TIFF generation.

Comment (mtbc) [reply contributing/data-model-schema.txt:194]: Is it just a case of using `tiffcomment` and `xsltproc`? Perhaps not.

**Comment by mtbc** [comment]
Thank you, good to merge.

File: contributing/data-model-schema.txt (hunk 1 above)
Comment (sbesson) [reply contributing/data-model-schema.txt:194]: Certainly sounds like a good starting point. Captured as https://trello.com/c/XyAzmbVg/435-ome-tiff-samples-upgrade-process

**Comment by dgault** [comment]
Document reads well and steps all look correct, links all appear to be working and cant find any further references to internal details. Looks good to merge from me too.

**Comment by hflynn** [comment]
I'll merge and release and then open a quick PR to add the link to the Model docs as mentioned above - thanks all

**Pull request merged**